mldsa/sign.c

```diff
@@ -254,8 +264,40 @@ int crypto_sign_signature(uint8_t *sig, size_t *siglen, const uint8_t *m,
     rnd[i] = 0;
 #endif
 
-  crypto_sign_signature_internal(sig, siglen, m, mlen, pre, 2 + ctxlen, rnd,
-                                 sk);
+  crypto_sign_signature_internal(sig, siglen, m, mlen, pre, 2 + ctxlen, rnd, sk,
+                                 0);
+  return 0;
+}
+
+/*************************************************
+ * Name:        crypto_sign_signature_extmu
+ *
+ * Description:  FIPS 204: Algorithm 2 ML-DSA.Sign external mu variant.
+ *               Computes signature.
+ *
+ * Arguments:   - uint8_t *sig:   pointer to output signature (of length
+ *CRYPTO_BYTES)
+ *              - size_t *siglen: pointer to output length of signature
+ *              - uint8_t *mu:    pointer to input mu to be signed
+ *              - size_t mulen:   length of mu
+ *              - uint8_t *sk:    pointer to bit-packed secret key
+ *
+ * Returns 0 (success) or -1 (context string too long)
+ **************************************************/
+int crypto_sign_signature_extmu(uint8_t *sig, size_t *siglen, const uint8_t *mu,
+                                size_t mulen, const uint8_t *sk)
```

Comment:
I don't understand why a mulen is needed? Isn't this always 64 bytes? 
The internal functiosn ignore it anyway. 
Can we remove `mulen` and replace `const uint8_t *mu` by `const uint8_t mu[CRHBYTES]`?

Reply:
Agreed - not used. [f8410a3](https://github.com/pq-code-package/mldsa-native/pull/47/commits/f8410a33d86a80b76089b473a9b242cd29bb9988)

Reply:
Thanks!